core/ir/instr_inline_api.h

```diff
@@ -156,7 +156,7 @@ opnd_is_predicate_reg(opnd_t op)
     return IF_AARCH64_ELSE(op.kind == REG_kind &&
                                op.value.reg_and_element_size.reg >= DR_REG_P0 &&
                                op.value.reg_and_element_size.reg <= DR_REG_P15,
-                           false);
+                           false && op.kind == REG_kind);
```

Comment:
Add a comment explaining why this is here (future readers are going to puzzle over this otherwise).

Reply:
Done.